extension-manifest-v3/src/background/dnr.js

```diff
@@ -74,21 +75,30 @@ if (__PLATFORM__ !== 'firefox') {
           enableRulesetIds,
           disableRulesetIds,
         });
-        console.info('DNR - Lists successfully updated');
+        console.info('DNR: lists successfully updated');
       } catch (e) {
-        console.error(`DNR - Error while updating lists:`, e);
+        console.error(`DNR: error while updating lists:`, e);
       }
     }
   });
 
+  const getPauseRulesIds = async () => {
+    return (await chrome.declarativeNetRequest.getDynamicRules())
+      .filter(({ id }) => id <= 3)
+      .map(({ id }) => id);
+  };
+
   observe('paused', async (paused, prevPaused) => {
-    // Skip if hostnames has not changed
+    // The background process starts and runs for each tab, so we can assume
+    // that this function is called before the user can change the paused state
+    // in the panel or the settings page.
     if (!prevPaused) return;
 
-    const dynamicRules = await chrome.declarativeNetRequest.getDynamicRules();
     const hostnames = Object.keys(paused);
 
     if (hostnames.length) {
+      const removeRuleIds = await getPauseRulesIds();
```

Comment:
can we move it above the if?

Reply:
It was here when I though that we must call it always (not check `prevPaused`). However, it is called for both cases, so for now it's pointless. Good catch.